faktory/exceptions.py

```diff
@@ -1,13 +1,26 @@
-__ALL__ = ['FaktoryHandshakeError', 'FaktoryAuthenticationError']
+__ALL__ = ["FaktoryHandshakeError", "FaktoryAuthenticationError", "FaktoryError"]
 
 
-class FaktoryHandshakeError(ConnectionError):
+class FaktoryError(Exception):
+    """
+    The base Faktory Exception that all other Faktory related
+    exceptions inherit from. To catch any other, more specific,
+    Faktory exception, catch this one and handle appropriately.
+
+    This Exception should not be used, and is in place strictly
+    as a way to catch all `Faktory` exceptions.
+    """
+
+    pass
+
+
+class FaktoryHandshakeError(FaktoryError, ConnectionError):
     pass
 
 
-class FaktoryAuthenticationError(ConnectionError):
+class FaktoryAuthenticationError(FaktoryError, ConnectionError):
     pass
 
 
-class FaktoryConnectionResetError(ConnectionResetError):
+class FaktoryConnectionResetError(FaktoryError, ConnectionError):
```

Comment:
Should this be `ConnectionResetError` rather than `ConnectionError` here?

Reply:
Yep absolutely should be! Good catch. Will make the change.